### PR TITLE
Update developer doc

### PR DESCRIPTION
**Pull request by amarts**
To reflect the latest changes in the project, and guidelines. This is not yet complete, but brings some level of sanity. Needs another round of polishing.

### CODE REVIEW

**Comment by prashanthpai** [comment]
@amarts Can you please fix the merge conflicts ?

File: Developer-guide/Backport-Guidelines.md [hunk 1]
```diff
@@ -1,87 +1,26 @@
-Bugs often get fixed in master before release branches. When a bug is
-fixed in the master branch, it might be desirable or necessary in a
-stable branch. To put the fix in the stable branch we need to backport the
-fix to the stable branch.
+As a policy, any new change, bug fix, etc., are to be fixed in 'master'
```
Comment (sankarshanmukhopadhyay) [inline Developer-guide/Backport-Guidelines.md]: In the GlusterFS project, as a policy, any new change, bug fix etc are to be fixed in 'master'

File: Developer-guide/Backport-Guidelines.md [hunk 2]
```diff
@@ -1,87 +1,26 @@
...
 
-This page describes the steps needed to backport simple changes. Changes
-that do not apply cleanly will need some manual modifications and using
-`git cherry-pick` may not always be the easiest solution.
+## Policy
 
-## via Git command-line
-1.  Git clone the GlusterFS code
+* No feature from master would be backported to the release branch
+* Any CVE (aka, security alerts) on the existing releases would be backported, after getting fixed in master branch.
```
Comment (sankarshanmukhopadhyay) [inline Developer-guide/Backport-Guidelines.md]: CVE ie. security vulnerability (listed on the CVE database)

File: Developer-guide/Backport-Guidelines.md [hunk 3]
```diff
@@ -1,87 +1,26 @@
...
+* Only severe issues, crashes, data loss, or unavailability issues would be backported to the release.
```
Comment (sankarshanmukhopadhyay) [inline Developer-guide/Backport-Guidelines.md]: Only topics which bring about data loss or, unavailability would be backported to the release.

File: Developer-guide/Backport-Guidelines.md [hunk 4]
```diff
@@ -1,87 +1,26 @@
...
+* For any other issues, upgrading to the higher version where bug is fixed is the recommended way.
+  - Gluster provides 'rolling' upgrade support, i.e., one can upgrade their version without stopping the application I/O, so we recommend migrating to higher version.
 
-                git clone ssh://username@review.gluster.org/glusterfs
+## Things to take care while backporting a patch.
 
-2.  Create and checkout a new branch for your work, based on the branch
-    for the backport version
+If your patch meets the criteria above, or you are an user, who prefer to have a fix backported, because your current setup is facing issues, below are the steps you need to take care to submit a patch on release branch.
```
Comment (sankarshanmukhopadhyay) [inline Developer-guide/Backport-Guidelines.md]: a user

File: Developer-guide/Backport-Guidelines.md [hunk 5]
```diff
@@ -1,87 +1,26 @@
...
-that do not apply cleanly will need some manual modifications and using
-`git cherry-pick` may not always be the easiest solution.
+## Policy
 
-## via Git command-line
-1.  Git clone the GlusterFS code
+* No feature from master would be backported to the release branch
+* Any CVE (aka, security alerts) on the existing releases would be backported, after getting fixed in master branch.
+* Only severe issues, crashes, data loss, or unavailability issues would be backported to the release.
+* For any other issues, upgrading to the higher version where bug is fixed is the recommended way.
```
Comment (sankarshanmukhopadhyay) [inline Developer-guide/Backport-Guidelines.md]: For any other issues, the project recommends that the installation be upgraded to a newer release where the specific bug has been addressed.

File: Developer-guide/Backport-Guidelines.md [hunk 6]
```diff
@@ -1,87 +1,26 @@
...
+  - Gluster provides 'rolling' upgrade support, i.e., one can upgrade their version without stopping the application I/O, so we recommend migrating to higher version.
 
-                git clone ssh://username@review.gluster.org/glusterfs
+## Things to take care while backporting a patch.
```
Comment (sankarshanmukhopadhyay) [inline Developer-guide/Backport-Guidelines.md]: to pay attention to while backporting a patch.

File: Developer-guide/Backport-Guidelines.md (hunk 2 above)
Comment (amarts) [reply Developer-guide/Backport-Guidelines.md]: Done

File: Developer-guide/Backport-Guidelines.md (hunk 3 above)
Comment (amarts) [reply Developer-guide/Backport-Guidelines.md]: Done

File: Developer-guide/Backport-Guidelines.md (hunk 1 above)
Comment (amarts) [reply Developer-guide/Backport-Guidelines.md]: Done

File: Developer-guide/Backport-Guidelines.md (hunk 6 above)
Comment (amarts) [reply Developer-guide/Backport-Guidelines.md]: Done

File: Developer-guide/Backport-Guidelines.md (hunk 5 above)
Comment (amarts) [reply Developer-guide/Backport-Guidelines.md]: Done

File: Developer-guide/Backport-Guidelines.md (hunk 4 above)
Comment (amarts) [reply Developer-guide/Backport-Guidelines.md]: Done

**Comment by amarts** [comment]
Addressed all the comments by @sankarshanmukhopadhyay 

@prashanthpai @nigelbabu can we merge this?

File: Developer-guide/Backport-Guidelines.md [hunk 7]
```diff
@@ -1,87 +1,32 @@
...
-1.  Git clone the GlusterFS code
+* No feature from master would be backported to the release branch
+* CVE ie., security vulnerability (listed on the CVE database) reported in
+the existing releases would be backported, after getting fixed in master
+branch.
+* Only topics which bring about data loss or, unavailability would be
+backported to the release.
+* For any other issues, the project recommends that the installation be
+upgraded to a newer release where the specific bug has been addressed.
+  - Gluster provides 'rolling' upgrade support, i.e., one can upgrade their
```
[the remark this answers is not in the document]
Comment (amarts) [reply Developer-guide/Backport-Guidelines.md]: No, not for client.

File: Developer-guide/Backport-Guidelines.md [hunk 8]
```diff
@@ -1,87 +1,32 @@
...
 
-This page describes the steps needed to backport simple changes. Changes
-that do not apply cleanly will need some manual modifications and using
-`git cherry-pick` may not always be the easiest solution.
+## Policy
 
-## via Git command-line
-1.  Git clone the GlusterFS code
+* No feature from master would be backported to the release branch
+* CVE ie., security vulnerability (listed on the CVE database) reported in
```
[the remark this answers is not in the document]
Comment (sankarshanmukhopadhyay) [reply Developer-guide/Backport-Guidelines.md]: I link like https://cve.mitre.org/cve/search_cve_list.html might be good to have.

File: Developer-guide/Projects.md [hunk 9]
```diff
@@ -1,28 +1,35 @@
...
 
-### gfsck - A GlusterFS filesystem check
+### glusterd2
 
--   A tool to check filesystem integrity and repairing
--   I'm currently working on it
--   Owner: Xavier Hernandez (Datalab)
+Repo: https://github.com/gluster/glusterd2
+
+New management layer for glusterfs. Has navitive ReST API, and uses etcd to
```
Comment (prashanthpai) [inline Developer-guide/Projects.md]: native ?

File: Developer-guide/Development-Workflow.md [hunk 10]
```diff
@@ -187,26 +157,13 @@ generate the build configuration:
...
+ sh$ make install
 
 #### GlusterFS UFO/SWIFT
 
 To build and run Gluster UFO you can do the following:
 
-1.  Build, create, and install the RPMS as described in
-    [CompilingRPMS](./Compiling-RPMS.md).
-2.  Configure UFO/SWIFT as described in [Howto Using UFO SWIFT - A quick
+1.  Configure UFO/SWIFT as described in [Howto Using UFO SWIFT - A quick
```
Comment (prashanthpai) [inline Developer-guide/Development-Workflow.md]: The link can be updated to this one: https://github.com/gluster/gluster-swift/blob/master/doc/markdown/quick_start_guide.md

File: Developer-guide/Backport-Guidelines.md [hunk 11]
```diff
@@ -1,87 +1,32 @@
...
-stable branch. To put the fix in the stable branch we need to backport the
-fix to the stable branch.
+In GlusterFS project, as a policy, any new change, bug fix, etc., are to be
+fixed in 'master' branch before release branches. When a bug is fixed in
+the master branch, it might be desirable or necessary in release branch.
 
-Anyone in the community can suggest a backport. If you are interested to
-suggest a backport, please check the [Backport
-Wishlist](./Backport-Wishlist.md).
+This page describes the policy GlusterFS has regarding the backports. As a user, or contributor, being aware of this policy would help you to understand how to request for backport from community.
```
Comment (prashanthpai) [inline Developer-guide/Backport-Guidelines.md]: It's better to stick to 80 character limit wherever possible - here and elsewhere in the PR. The hyperlinks are an exception though. Markdown doesn't interpret single line break as a new line during rendering.

File: Developer-guide/Projects.md (hunk 9 above)
Comment (amarts) [reply Developer-guide/Projects.md]: Done

File: Developer-guide/Development-Workflow.md (hunk 10 above)
Comment (amarts) [reply Developer-guide/Development-Workflow.md]: Moved it to another place, and updated the Link

File: Developer-guide/Backport-Guidelines.md (hunk 11 above)
Comment (amarts) [reply Developer-guide/Backport-Guidelines.md]: Done

**Pull request merged**